docs/reference/troubleshooting/troubleshooting-searches.asciidoc

```diff
@@ -0,0 +1,240 @@
+
+[[troubleshooting-searches]]
+== Troubleshooting searches
+
+When you query your data, Elasticsearch may return an error, no search results,
+or results in an unexpected order. This guide describes how to troubleshoot
+searches.
+
+[discrete]
+[[troubleshooting-searches-exists]]
+=== Ensure the data stream, index, or alias exists
+
+Elasticsearch returns an `index_not_found_exception` when the data stream, index
+or alias you try to query does not exist. This can happen when you misspell the
+name or when the data has been indexed to a different data stream or index.
+
+Use the <<indices-exists,exists API>> to check whether a data stream, index, or
+alias exists:
+
+[source,console]
+----
+HEAD my-data-stream
+----
+
+Use the <<data-stream-stats-api,data stream stats API>> to list all data
+streams:
+
+[source,console]
+----
+GET /_data_stream/_stats?human=true
+----
+
+Use the <<cat-indices,cat indices API>> to list all indices:
+
+[source,console]
+----
+GET _cat/indices?v=true&s=index&h=index,docs.count,status,health
+----
+
+Use the <<cat-alias,cat aliases API>> to list all aliases:
+
+[source,console]
+----
+GET _cat/aliases?v=true&s=alias&h=alias,index,is_write_index
+----
+
+[discrete]
+[[troubleshooting-searches-data]]
+=== Ensure the data stream or index contains data
+
+When a search request returns no hits, the data stream or index may contain no
+data. This can happen when there is a data ingestion issue. For example, the
+data may have been indexed to a data stream or index with another name.
+
+Use the <<search-count,count API>> to retrieve the number of documents in a data
+stream or index. Check that `count` in the response is not 0.
+
+////
+[source,console]
+----
+PUT my-index-000001
+----
+////
+
+[source,console]
+----
+GET /my-index-000001/_count
+----
+//TEST[continued]
+
+NOTE: When getting no search results in {kib}, check that you have selected the
+correct data view and a valid time range. Also, ensure the data view has been
+configured with the correct time field.
+
+[discrete]
+[[troubleshooting-searches-field-exists]]
+=== Ensure the field exists and has data
+
+Querying a field that does not exist or has no data will not return any results.
+Use the <<query-dsl-exists-query,`exists` query>> to check whether there are
+documents that contain an indexed value for a field. Check that `count` in the
+response is not 0.
+
+[source,console]
+----
+GET /my-index-000001/_count
+{
+  "query": {
+    "exists": {
+      "field": "my-field"
+    }
+  }
+}
+----
+//TEST[continued]
+
+If the field exists, and is not of type `text`, use a
+<<search-aggregations-bucket-terms-aggregation,terms aggregation>> to retrieve
+the field's most common values:
+
+[source,console]
+----
+GET /my-index-000001/_search?filter_path=aggregations
+{
+  "size": 0,
+  "aggs": {
+    "top_values": {
+      "terms": {
+        "field": "my-field",
+        "size": 10
+      }
+    }
+  }
+}
```

Comment:
would this apply to numeric fields too? Or would you rather use a different stats agg for numerics?

Reply:
I wasn't sure whether to include an agg in this guide. At this point we don't even know if the field is aggregatable. Maybe we add a separate section "Check for common values" after the field caps section and include a terms and a stats agg there?

Reply:
field_caps would tell you if the field is aggregatable. I am ok with your plan